src/nunavut/lang/cpp/templates/_fields_as_union.j2

```diff
@@ -3,17 +3,77 @@
  # Copyright (C) 2021  UAVCAN Development Team  <uavcan.org>
  # This software is distributed under the terms of the MIT License.
 #}
-    union // The union is placed first to ensure that the active element address equals the object address.
+    class VariantType final
     {
-{%- for field in composite_type.fields_except_padding -%}
-        {{ field.doc | block_comment('cpp-doxygen', 8, 120) }}
-        {{ field.data_type | declaration }} {{ field.name | id }};
+    public:
+        VariantType()
+            : tag_(0)
+            , internal_union_value_()
+        {}
+
+        //
+        // Rule of Five.
+        //
+        VariantType(const VariantType&) = delete;
+        VariantType& operator=(const VariantType&) = delete;
+        VariantType(VariantType&&) = delete;
+        VariantType& operator=(VariantType&&) = delete;
```

Comment:
This seems unnecessarily restrictive. Most if not all of the generated types will be POD types that can be safely moved and copied. Removing this possibility pretty much requires the use of dynamic storage for passing objects around the application, which is probably not something we want to require. Shall a non-copyable/non-movable member exist, the default copy/move constructors/operators would trigger a compile-time error anyway, so I don't expect us to lose anything by replacing your `delete` with `default` here.

Reply:
Okay. I'll have to write tests to cover this so I'll get to it tomorrow.

Reply:
Coded this up. I still beed to write tests and fix the python3.9 CI.